### PR TITLE
[java] CodeQL query, Increase fastjson detection. Improve RemoteFlowSource class, support SpringMvc.

### PR DESCRIPTION
**Pull request by haby0**
CodeQL query, Increase fastjson detection. Improve RemoteFlowSource class, support SpringMvc.

### CODE REVIEW

File: java/ql/src/semmle/code/java/frameworks/FastJson.qll [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import java
+
+
+/**
+ * The class `com.alibaba.fastjson.JSON`.
+ */
+class FastJson extends RefType {
+    FastJson() { 
+        this.hasQualifiedName("com.alibaba.fastjson", "JSON") or
```
Comment (intrigus-lgtm) [inline java/ql/src/semmle/code/java/frameworks/FastJson.qll:9]: I think you should probably either change this predicate or adjust the comment to include `JSONObject` as well.

File: java/ql/src/Security/CWE/CWE-502/UnsafeDeserialization.ql [hunk 2]
```diff
@@ -18,12 +18,14 @@ import DataFlow::PathGraph
...
+  }
 
   override predicate isSink(DataFlow::Node sink) { sink instanceof UnsafeDeserializationSink }
 }
 
 from DataFlow::PathNode source, DataFlow::PathNode sink, UnsafeDeserializationConfig conf
 where conf.hasFlowPath(source, sink)
 select sink.getNode().(UnsafeDeserializationSink).getMethodAccess(), source, sink,
-  "Unsafe deserialization of $@.", source.getNode(), "user input"
+ "Unsafe deserialization of $@.", source.getNode(), "user input"
```
Comment (intrigus-lgtm) [inline java/ql/src/Security/CWE/CWE-502/UnsafeDeserialization.ql:31]: This seems to not be autoformatted.
You can run the autoformatter from VS Code on all files.

File: java/ql/src/semmle/code/java/FlowSources.qll [hunk 3]
```diff
@@ -0,0 +1,281 @@
+/**
```
Comment (intrigus-lgtm) [inline java/ql/src/semmle/code/java/FlowSources.qll:1]: This feels seems to be a duplicate of the existing `FlowSources.qll` file.

Comment (haby0) [reply java/ql/src/semmle/code/java/FlowSources.qll:1]: I don’t think so. The ```FlowSources.qll``` file is missing for SpringMVC for remote input. The ```SpringServletInputParameterSource``` class can only get the remote part of the user's input,  for example the following code is in the ```source``` Won't get it.
```java
@GetMapping(value = "index")
public void index(String request){
     System.out.println(request);
}
```
I added the ```SpringMVC``` class to the ```FlowSources.qll``` file, mainly to verify whether the class annotation where the method parameters are obtained uses the ``Mapping``` and sub-annotations.
You can try the above code with the following statement.
```java
import java
import semmle.code.java.dataflow.FlowSources

from RemoteFlowSource rfs
select rfs
```

**Comment by aschackmull** [comment]
Looks like this is superseded by https://github.com/github/codeql/pull/3674. Closing.

**Pull request closed without merging**